allen-cell-animated/nucmorph-colorizer
# feat: Categorical color selection

Problem
=======
Continues work on #148! Adds the ability to select from palettes for categorical colors and color individual categories.

Solution
========
- `ColorRampDropdown` can now switch to show categorical palettes, instead of only color ramps. Some custom styling was added to add visual separation between each palette.
  - Shows a number of colors based on the number of categories present in the selected feature.
- Adds a varying number of labeled color pickers through the new `CategoricalColorPicker` component, which serves as both the legend and a way to customize color ramps.
- Updates the canvas tooltip to show categories.
- Reorders palettes and replaces duplicate colors.
- Adds additional categorical features to the example dataset, to show different numbers of categories.


## Type of change
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open this preview link: https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-153/?dataset=https%3A%2F%2Fraw.githubusercontent.com%2Fallen-cell-animated%2Fnucmorph-colorizer%2Ffeature%2Fcategorical-color-selection%2Fdocumentation%2Fexample_dataset%2Fmanifest.json&feature=Categorical%20Feature%20(5)&color=matplotlib-cool
1. Switch the feature between the three different categorical features. Different numbers of color pickers should be shown for each, and the number of colors in the displayed palette should also change.
1. Change the palette from the dropdown.
1. Click on a color picker and edit the color. You should see it change on the viewport and in the dropdown palette viewer.

Screenshots (optional):
-----------------------
![image](https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/952703b8-28ef-48cf-9b5d-620d412d13f3)

https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/6318b6aa-d77c-4cc1-9221-191f2e9a8614

Keyfiles (delete if not relevant):
-----------------------
1. `ColorRampDropdown.tsx`
3. `CategoricalColorPicker.tsx`
